blip-00xx.md

```diff
@@ -0,0 +1,220 @@
+encoding described by [BOLT 11][bolt11] which can be used to communicate 
+encoded blinded path information to the payer of the invoice. 
+
+## Copyright
+
+This bLIP is licensed under the CC0 license.
+
+## Rational
+
+Blinded paths have been included in the [BOLT 4 specification][blinded-paths] 
+and using them in the context of receiving payments is natively supported in 
+the [BOLT 12 Offers proposal][offers]. However, the Offers proposal also 
+includes various other dependencies such as onion messaging and various new 
+protocol message all of which will require a network wide upgrade before full 
+advantage can be taken of the new protocol. Blinded paths themselves are a 
+useful tool for privacy and potentially more reliable payment delivery and so 
+this document proposes a carve-out to the existing [BOLT 11] invoice format so 
+that advantage can be taken of the blinded paths protocol in the context of 
+payments in a sooner time frame. This will be done by adding a new tagged-field 
+type to the BOLT 11 invoice specification that will encode a blinded payment 
+path. With this bLIP, the sender and the receiver of a payment will need to be 
+aware of the new tagged-field and the receiver will need to support route 
+blinding and ideally have direct channel peers who also support route blinding 
+in order to start widening their anonymity set.
+
+## Specification
+
+### Tagged Field
+
+The proposal is to add the following new [tagged field][tagged-fields] to the 
+set defined in [BOLT 11][bolt11]:
+
+- `b` (20): `data_length` variable. One or more entries each containing a 
+   blinded payment path for a private route; there may be more than one `b` 
+   field. The field uses the `blinded_payinfo` type described below which draws 
+   heavily on the proposed encoding of the `blinded_payinfo` subtype defined in 
+   the [Offers proposal][offers].
+
+1. subtype: `blinded_payinfo`
+2. data:
+  * [`u32`:`fee_base_msat`]
+  * [`u32`:`fee_proportional_millionths`]
+  * [`u16`:`cltv_expiry_delta`]
+  * [`u64`:`htlc_minimum_msat`]
+  * [`u64`:`htlc_maximum_msat`]
+  * [`u16`:`flen`]
+  * [`flen*byte`:`features`]
+  * [`33*byte`:`first_ephemeral_blinding_point`]
+  * [`byte`:`num_hops`]
+  * [`num_hops*blinded_hop`:`blinded_hops`]
+
+1. subtype: `blinded_hop`
+2. data:
+   * [`33*byte`:`blinded_node_pubkey`]
+   * [`2*byte`:`cipher_text_length`]
+   * [`cipher_text_length*byte`:`cipher_text`]
+
+The `blinded_node_pubkey` of the first `blinded_hop` in a `blinded_payinfo` is
+the real public key of the blinded path's introduction node. 
+
+### Requirements
+
+An invoice containing the `b` field type:
+- MUST not contain the `r` field type.
+- MUST not contain the `s` field type since a payment address in the context of
+  blinded payments does not make sense since the recipient is able to use the
+  `path_id` in the `encrypted_recipient_data` for the same purpose.
+- SHOULD sign the invoice with a private key that is not the same as their 
```

Comment:
well we do have the random key in the "destination" field. so we can atleast verify that "invoice hasnt been changed since original creation", ie, no-one was able to take the invoice that looks like it is coming from you but then quickly sneak their own blinded paths to themselves in it. 

Perhaps could do something fancier like: the sig is the `n`-of-`n` musig2 combo created by combining sigs from _all_ the final blinded pubkeys in the `n` blinded paths the invoice contains ...?

Reply:
Hypothetically you could just craft an entirely new invoice though and sign it with a random privkey. So maybe it guards against a super naive tampering but even a modestly intentional tampering is still possible, no? Maybe I'm not understanding completely (I haven't read BOLT11 end-to-end).

Reply:
Yeah this came up in the spec meeting the other week. I think some plan to use this key in a sort of "address book" entry. It doesn't need to be your _actual_ node's public key, and you can rotate it, but it can be useful as a pseudonym on the network.

Reply:
The signature lets us verify the integrity of the invoice (nothing has been tampered with). As for authentication, when dealing with hop hints, it's already the case today that the public key included won't be in the public graph, so you don't have a way to verify that the node actually "exists".